### ISSUE TITLE
Ephemeral: sandpaper::serve() can accidentally change local git config to bot

### ISSUE THREAD

**Issue by zkamvar** [carpentries/sandpaper#449]
I have no clue how this works, but there are cases where someone uses sandpaper::serve() locally and the Carpentries Bot is set as the git user: https://github.com/joelnitta/targets-workshop/commit/7ebe2f7c5e575e7b1cf181ef3302b268b1e99368

I ran into this issue exactly once today where `.git/config` was amended to include:

```
[user]
   name: "Carpenter"
   email: "team@carpentries.org"
```

I wasn't allowed to commit because I do not have a signing key for that account and I sign all of my commits. 

removing those lines restored my default commit status, but I now am faced with the issue of not knowing _how_ this was amended.

## Source

The code that sets the carpenter if no user is set is: 

https://github.com/carpentries/sandpaper/blob/3d088c7fe81191d665cb8a812aff737e5c1b145f/R/utils-git.R#L295-L302

It's also used in two places:

https://github.com/carpentries/sandpaper/blob/3d088c7fe81191d665cb8a812aff737e5c1b145f/R/build_paths.R#L10-L19

Which is used in `create_lesson()`

and 

https://github.com/carpentries/sandpaper/blob/3d088c7fe81191d665cb8a812aff737e5c1b145f/R/create_site.R#L1-L21

which gets called every time a lesson is built when the `site/` folder does not contain a `DESCRIPTION` file. This calls `create_description()`, which uses `gert::git_signature_default(path)` to get the author name:

https://github.com/carpentries/sandpaper/blob/3d088c7fe81191d665cb8a812aff737e5c1b145f/R/utils.R#L156-L171

In hindsight, it's not necessary to do this because the DESCRIPTION is one of the ephemeral pieces we use in the build process that is not propogated to the site or the cache; it's merely an anchor for {pkgdown} to grab on to.

**Comment by zkamvar** [comment]
Also related is: https://github.com/carpentries-incubator/statistical-probabilistic-programming-r/commit/9fbf6c44e1246afc0b09a9799d5bba570ef7229f

**Comment by zkamvar** [comment]
I believe this was fixed by #476, so I will close it.
